Plugins/JiraCommitHintPlugin/JiraCommitHintPlugin.csproj

```diff
@@ -63,6 +64,10 @@
       <Project>{0F1F1168-A4B2-4FA2-B17B-735140D17F39}</Project>
       <Name>GitExtUtils</Name>
     </ProjectReference>
+    <ProjectReference Include="..\..\GitUI\GitUI.csproj">
```

Comment:
Plugins must not reference any projects, that introduces coupling.

Reply:
I did it because I found other plugins with this reference. Also I planned that new CredentialsControl will be used for Bitbucket plugin, because this plugin also stores credential as plain text. What assembly should I use to implement common controls?

Reply:
Make sense, however GitUI is only used by the main app. Plugins should not be coupled to this project. I believe plugins current have two projects that could couple to - GitUIPluginInterfaces and ResourceManager, so anything plugin-purpose should be placed in these.

Please note we are working on decoupling and moving out plugins from the solution and the main repo, and changes like this would make that work harder.